core/src/test/java/com/sequenceiq/cloudbreak/service/stack/CloudResourceAdvisorTest.java

```diff
@@ -0,0 +1,83 @@
+import com.sequenceiq.cloudbreak.domain.Blueprint;
+import com.sequenceiq.cloudbreak.service.blueprint.BlueprintService;
+import com.sequenceiq.cloudbreak.service.blueprint.BlueprintTextProcessorFactory;
+import com.sequenceiq.cloudbreak.template.processor.BlueprintTextProcessor;
+import com.sequenceiq.cloudbreak.workspace.model.Workspace;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CloudResourceAdvisorTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CloudResourceAdvisorTest.class);
+
+    private static final String VERSION_7_2_1 = "7.2.1";
+
+    private static final String VERSION_7_2_0 = "7.2.0";
+
+    private static final String TEST_BLUEPRINT_NAME = "testBp";
+
+    private static final String TEST_CLUSTER_CRN = "crn:cdp:datahub:us-west-1:autoscale:cluster:f7563fc1-e8ff-486a-9260-4e54ccabbaa0";
+
+    @Mock
+    private StackService stackService;
+
+    @Mock
+    private Workspace workspace;
+
+    @Mock
+    private BlueprintTextProcessor blueprintTextProcessor;
+
+    @InjectMocks
+    private CloudResourceAdvisor underTest;
+
+    @Mock
+    private BlueprintService blueprintService;
+
+    @Mock
+    private Blueprint blueprint;
+
+    @Mock
+    private BlueprintTextProcessorFactory blueprintTextProcessorFactory;
+
+    @Mock
+    private CmTemplateProcessorFactory cmTemplateProcessorFactory;
+
+    @Test
+    public void testRecommendAutoscaleWhenCloudManagerVersionLessThanEqualTo720() {
+        when(blueprintTextProcessor.getVersion()).thenReturn(java.util.Optional.of(VERSION_7_2_0));
+        Blueprint blueprint = new Blueprint();
+        blueprint.setBlueprintText("{\"Blueprints\":{123:2}}");
+        when(blueprintTextProcessorFactory.createBlueprintTextProcessor("{\"Blueprints\":{123:2}}")).thenReturn(blueprintTextProcessor);
+        when(blueprintService.getByNameForWorkspaceId(any(), anyLong())).thenReturn(blueprint);
+        assertEquals(new AutoscaleRecommendation(Set.of(), Set.of()),
+                underTest.getAutoscaleRecommendation(workspace.getId(), TEST_BLUEPRINT_NAME));
+    }
+
+    @Test
+    public void testRecommendAutoscaleWhenCloudManagerVersionGreaterThanEqualTo721() {
+        when(blueprintTextProcessor.getVersion()).thenReturn(Optional.of(VERSION_7_2_1));
+        Blueprint blueprint = new Blueprint();
+        blueprint.setBlueprintText("{\"Blueprints\":{123:2}}");
+        when(blueprintTextProcessorFactory.createBlueprintTextProcessor("{\"Blueprints\":{123:2}}")).thenReturn(blueprintTextProcessor);
+        when(blueprintTextProcessor.recommendAutoscale()).thenReturn(new AutoscaleRecommendation(Set.of("nonempty"), Set.of("nonempty")));
+        when(blueprintService.getByNameForWorkspaceId(any(), anyLong())).thenReturn(blueprint);
+        assertEquals(new AutoscaleRecommendation(Set.of("nonempty"), Set.of("nonempty")),
+                underTest.getAutoscaleRecommendation(workspace.getId(), TEST_BLUEPRINT_NAME));
```

Comment:
Replace "nonempty" with "compute" so that it is easy to understand that this is a real hostGroup